controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1358,6 +1360,109 @@ func (r *BareMetalHostReconciler) actionDeprovisioning(prov provisioner.Provisio
 	return actionComplete{}
 }
 
+func (r *BareMetalHostReconciler) checkServicing(prov provisioner.Provisioner, info *reconcileInfo, hup *metal3api.HostUpdatePolicy) (result actionResult, isServicing bool) {
+	servicingData := provisioner.ServicingData{}
+	// start janders mods
+	var fwDirty bool
+	var hfsDirty bool
+
+	// (NOTE)janders: since Servicing is an opt-in feature that requires HostUpdatePolicy to be created and set to onReboot
+	// set below booleans to false by default and change to true based on policy settings
+
+	servicingData.LiveFirmwareUpdateAllowed = false
+	servicingData.LiveFirmwareSettingsAllowed = false
+
+	info.log.Info(fmt.Sprintf("janders-debug: in checkServicing(). Defaults: LiveFirmwareUpdateAllowed: %+v LiveFirmwareSettingsAllowed:  %+v. (end of message)", servicingData.LiveFirmwareUpdateAllowed, servicingData.LiveFirmwareSettingsAllowed))
+	if hup == nil {
+		info.log.Info(fmt.Sprintf("janders-debug: in checkServicing(). HostUpdatePolicy is nil"))
+	}
+	if hup != nil {
+		info.log.Info(fmt.Sprintf("janders-debug: in checkServicing(). HostUpdatePolicy value:  %+v", hup))
+		if hup.Spec.FirmwareSettings == metal3api.HostUpdatePolicyOnReboot {
+			servicingData.LiveFirmwareSettingsAllowed = true
+			info.log.Info(fmt.Sprintf("janders-debug: in checkServicing(). hup.Spec.FirmwareSettings set to OnReboot, setting LiveFirmwareSettingsAllowed to  %+v", servicingData.LiveFirmwareSettingsAllowed))
+		}
+		if hup.Spec.FirmwareUpdates == metal3api.HostUpdatePolicyOnReboot {
+			servicingData.LiveFirmwareUpdateAllowed = true
+			info.log.Info(fmt.Sprintf("janders-debug: in checkServicing(). hup.Spec.FirmwareUpdates set to OnReboot, setting LiveFirmwareUpdateAllowed to  %+v", servicingData.LiveFirmwareUpdateAllowed))
+		}
+	}
+
+	if servicingData.LiveFirmwareUpdateAllowed {
+		if !reflect.DeepEqual(info.host.Status.Provisioning.Firmware, info.host.Spec.Firmware) {
+			servicingData.FirmwareConfig = info.host.Spec.Firmware
+			fwDirty = true
+			info.log.Info(fmt.Sprintf("janders-debug: setting fwDirty to  %+v", fwDirty))
+		}
+	}
+
+	if servicingData.LiveFirmwareSettingsAllowed {
+		hfsDirty, hfs, err := r.getHostFirmwareSettings(info)
```

Comment:
I have a wild guess that Go is being stupid here and creates a new locally scoped variable for hfsDirty. What if you do this instead:

```go
var hfs *metal3api.HostFirmwareSettings
hfsDirty, hfs, err = r.getHostFirmwareSettings(info)
```

Important: use `=` not `:=`!!!